samples/boards/stm32/backup_sram/src/main.c

```diff
@@ -24,14 +31,19 @@ int main(void)
 		return 0;
 	}
 
-	printk("Current value in backup SRAM (%p): %d\n", &backup_value, backup_value);
+	if (backup.magic != BACKUP_MAGIC) {
+		backup.magic = BACKUP_MAGIC;
+		backup.value = 0;
+	}
+
+	printk("Current value in backup SRAM (%p): %d\n", &backup.value, backup.value);
 
-	backup_value++;
+	backup.value++;
 #if __DCACHE_PRESENT
-	SCB_CleanDCache_by_Addr(&backup_value, sizeof(backup_value));
+	SCB_CleanDCache_by_Addr(&backup.value, sizeof(backup.value));
```

Comment:
Shouldn't the D$ for the whole structure be cleaned instead, since we might also write to `backup.magic`?
```suggestion
	SCB_CleanDCache_by_Addr(&backup, sizeof(backup));
```